addon/index.js

```diff
@@ -145,106 +141,50 @@ function initServerEnvironmentPreference() {
     if (lastVisitedName && lastVisitedName !== currName) {
       // Switch to the last visited environment.
       simplePrefs.prefs.SERVER_ENVIRONMENT = lastVisitedName;
+      updatePrefs();
+    }
+
+    if (self.loadReason === 'install') {
+      openOnboardingTab();
     }
 
     // Finally, watch for pref changes, kick off the env setup.
     simplePrefs.on('SERVER_ENVIRONMENT', updatePrefs);
-    updatePrefs();
   });
 }
 
 function openOnboardingTab() {
-  // TODO: Settings is populated async, so we need a global afterSettingsReady
-  // promise that this function can wait on.
   tabs.open({
-    url: SERVER_ENVIRONMENTS.production.BASE_URL + '/onboarding',
+    url: settings.BASE_URL + '/onboarding',
     inBackground: true
   });
 }
 
-function setupApp() {
-  updateExperiments().then(() => {
-    app = new Router(messageBridgePageMod);
-
-    app.on('uninstall-self', uninstallSelf);
-
-    app.on('install-experiment', installExperiment);
-
-    app.on('uninstall-experiment', uninstallExperiment);
-
-    app.on('sync-installed', () => {
-      app.send('sync-installed-result', {
-        clientUUID: store.clientUUID,
-        installed: store.installedAddons
-      });
-    });
-
-    if (self.loadReason === 'install') {
-      app.send('addon-self:installed');
-    } else if (self.loadReason === 'enable') {
-      app.send('addon-self:enabled');
-    } else if (self.loadReason === 'upgrade') {
-      app.send('addon-self:upgraded');
-    }
-  });
-}
-
-function Router(mod) {
-  this.mod = mod;
-  this._events = {};
-  this.mod.port.on('from-web-to-addon', function(evt) {
-    if (this._events[evt.type]) this._events[evt.type](evt.data);
-  }.bind(this));
-  return this;
-}
-
-Router.prototype.on = function(name, f) {
-  this._events[name] = f;
-  return this;
-};
-
-Router.prototype.send = function(name, data) {
-  this.mod.port.emit('from-addon-to-web', {type: name, data: data});
-  return this;
-};
-
-function updateExperiments() {
-  // Fetch the list of available experiments
-  return requestAPI({
-    url: settings.BASE_URL + '/api/experiments'
-  }).then(res => {
-    // Index the available experiments by addon ID
-    store.availableExperiments = {};
-    res.json.results.forEach(exp => {
-      store.availableExperiments[exp.addon_id] = exp;
-    });
-
-    // Update UI since we just updated experiments.
-    ExperimentNotifications.maybeSendNotifications();
-    ToolbarButton.updateButtonBadge();
-
-    // Query all installed addons
-    return new Promise(resolve => AddonManager.getAllAddons(resolve));
-  }).then(addons => {
+function experimentsLoaded(experiments) {
```

Comment:
This replaces the second part of `updateExperiments` its triggered by the `app` 'loaded' event.